x-pack/plugins/apm/server/lib/security/get_indices_privileges.ts

```
@@ -3,22 +3,34 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { Logger } from 'src/core/server';
 import { Setup } from '../helpers/setup_request';
+import { IndexPrivileges } from '../helpers/es_client';
 
-export async function getIndicesPrivileges(setup: Setup) {
+export async function getIndicesPrivileges(
+  setup: Setup,
+  logger: Logger
+): Promise<IndexPrivileges> {
   const { client, indices } = setup;
-  const response = await client.hasPrivileges({
-    index: [
-      {
-        names: [
-          indices['apm_oss.errorIndices'],
-          indices['apm_oss.metricsIndices'],
-          indices['apm_oss.transactionIndices'],
-          indices['apm_oss.spanIndices']
-        ],
-        privileges: ['read']
-      }
-    ]
-  });
-  return response.index;
+  try {
+    const response = await client.hasPrivileges({
+      index: [
+        {
+          names: [
+            indices['apm_oss.errorIndices'],
+            indices['apm_oss.metricsIndices'],
+            indices['apm_oss.transactionIndices'],
+            indices['apm_oss.spanIndices']
+          ],
+          privileges: ['read']
+        }
+      ]
+    });
+    return response;
+  } catch (error) {
+    if (error.statusCode === 400) {
```

Add a comment here explaining why the error is being swallowed.